src/julia_atomics.h

```diff
@@ -73,7 +73,11 @@ enum jl_memory_order {
  * are). We also need to access these atomic variables from the LLVM JIT code
  * which is very hard unless the layout of the object is fully specified.
  */
-#define jl_fence() atomic_thread_fence(memory_order_seq_cst)
+#if defined(_CPU_X86_64_)
```

Comment:
Can you check the GCC compiler version here?

Reply:
For what versions of gcc should we inline the locked instruction then?

Reply:
GCC < 11

Reply:
If we're inlining the same instructions from linux, then I think it might make sense to inline it for gcc > 11 as well: IIRC gcc uses the less efficient orq on %rsp (rather than doing it at a negative offset).

Reply:
I am not sure we can use the Linux version. ASAN might get very angry.